include/aws/common/string.h

```diff
@@ -58,31 +58,75 @@ struct aws_string {
 #endif
 
 /**
- * Equivalent to str->bytes. Here for legacy reaasons.
+ * Evaluates the set of properties that define the shape of all valid aws_string structures.
+ * It is also a cheap check, in the sense it run in constant time (i.e., no loops or recursion).
+ */
+AWS_STATIC_IMPL bool aws_string_is_valid(const struct aws_string *str) {
+    return str && AWS_MEM_IS_READABLE(&str->bytes[0], str->len + 1) && str->bytes[str->len] == 0;
+}
+
+/**
+ * Best-effort checks aws_string invariants, when the str->len is unknown
+ */
+AWS_STATIC_IMPL bool aws_c_string_is_valid(const struct aws_string *str) {
```

Comment:
It looks like this function should accept a `const char *`, right? At least, every time it's called that's what it's given.

Reply:
Correct! Done!